imxrt-hal/src/can/mod.rs

```diff
@@ -0,0 +1,994 @@
+                    1 << 15
+                } else {
+                    0
+                } & 0xFFFF;
+                mask |= (1 << 30) | (1 << 14);
+                let filter: u32 = (if ide == filter::FlexCanIde::Ext { 1 } else { 0 } << 30)
+                    | (if ide == filter::FlexCanIde::Ext { 1 } else { 0 } << 14)
+                    | (if remote == filter::FlexCanIde::Rtr {
+                        1
+                    } else {
+                        0
+                    } << 31)
+                    | (if remote == filter::FlexCanIde::Rtr {
+                        1
+                    } else {
+                        0
+                    } << 15)
+                    | (if ide == filter::FlexCanIde::Ext {
+                        (id >> (29 - 14)) << 16
+                    } else {
+                        (id & 0x7FF) << 19
+                    })
+                    | (if ide == filter::FlexCanIde::Ext {
+                        (id >> (29 - 14)) << 0
+                    } else {
+                        (id & 0x7FF) << 3
+                    });
+                this.write_mailbox_idflt_tab(filter_id, Some(filter));
+                let offset = this.mailbox_offset();
+                if filter_id < offset.clamp(0, 32) as u8 {
+                    this.write_mailbox_rximr(filter_id, Some(mask));
+                }
+                write_reg!(ral::can, this.reg, RXFGMASK, 0x7FFF7FFF);
+            }
+            ral::can::MCR::IDAM::RW::IDAM_2 => {
+                // not implemented
+            }
+            ral::can::MCR::IDAM::RW::IDAM_3 => {
+                // not implemented
+            }
+            _ => {}
+        })
+    }
+
+    pub fn set_fifo_interrupt(&mut self, enabled: bool) {
+        /* FIFO must be enabled first */
+        if !self.fifo_enabled() {
+            return;
+        };
+        /* FIFO interrupts already enabled */
+        const FLEXCAN_IMASK1_BUF5M: u32 = 0x00000020;
+        if (ral::read_reg!(ral::can, self.reg, IMASK1, BUFLM) & FLEXCAN_IMASK1_BUF5M) != 0 {
+            return;
+        };
+        /* disable FIFO interrupt flags */
+        modify_reg!(ral::can, self.reg, IMASK1, |reg| reg & !0xFF);
+        /* enable FIFO interrupt */
+        if enabled {
+            const FLEXCAN_IMASK1_BUF5M: u32 = 0x00000020;
+            modify_reg!(ral::can, self.reg, IMASK1, BUFLM: FLEXCAN_IMASK1_BUF5M);
+        }
+    }
+
+    fn fifo_enabled(&self) -> bool {
+        ral::read_reg!(ral::can, self.reg, MCR, RFEN == RFEN_1)
+    }
+
+    fn mailbox_offset(&self) -> u8 {
+        if self.fifo_enabled() {
+            let max_mailbox = self.get_max_mailbox() as u32;
+            let num_rx_fifo_filters = (read_reg!(ral::can, self.reg, CTRL2, RFFN) + 1) * 2;
+            let remaining_mailboxes = max_mailbox - 6_u32 - num_rx_fifo_filters;
+            /* return offset MB position after FIFO area */
+            if max_mailbox < max_mailbox - remaining_mailboxes {
```

Comment:
clippy impressed me with this one. Is there a different way to express this mailbox offset?

```
warning: you are trying to use classic C underflow conditions that will fail in Rust
   --> imxrt-hal/src/can/mod.rs:722:16
    |
722 |             if max_mailbox < max_mailbox - remaining_mailboxes {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#overflow_check_conditional
    = note: `#[warn(clippy::overflow_check_conditional)]` on by default
```